client/src/lib/ApiClient.js

```diff
@@ -76,6 +76,13 @@ const apiClient = {
       .then(callback)
       .catch(logError);
   },
+  createMetric: function(metricObj, callback) {
+    return axios
+      .post('/api/metrics', metricObj)
+      .then(unwrapData)
+      .then(callback)
+      .catch(logError);
+  },
   editMetric: function(id, updatedFields, callback) {
```

Comment:
When we edit a metric's query string, what happens to the data and analysis we've already done on existing experiments? Should we allow this?

Reply:
I think the only use case they'd have for editing the query string is if their data schema changes for some reason, which should be extremely rare. We know that the previous query was valid though, so I don't see a need to change the old data we collected. We can just start using the new query string from the day they update it and it shouldn't change the analysis.